libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Resources/ResourceExplorer.cs

```diff
@@ -639,13 +636,13 @@ private void ResourceProvider_Changed(object sender, IEnumerable<Resource> resou
             }
         }
 
-        private void AutoAssignId(Resource resource, JToken jToken)
+        private void AutoAssignId(Resource resource, JToken jToken, SourceContext sourceContext)
```

Comment:
>AutoAssignId [](start = 21, length = 12)

can we add a unit test to verify that sourceRange for an object matches to what we think it should be?

Reply:
I might need to add internal methods and add InternalsVisibleTo for that, since all of this is private.

Reply:
Added this test plus a few more to cover all combinations